algebra/vector_space.lean

```diff
@@ -0,0 +1,305 @@
+import algebra.ring
+import algebra.module
+
+universes u' v'
+
+class vector_space (α : inout Type u')  (β : Type v') [field α] extends module α β
+
+instance field.to_vector_space {K : Type} [field K] :
+vector_space K K := ⟨ring.to_module⟩
+
+lemma eq_zero_of_add_self_eq {α : Type} [add_group α]
+{a : α} (H : a + a = a) : a = 0 :=
+add_left_cancel (by {rw add_zero, exact H})
+
+class subspace (K : Type) (V : Type) [field K] [vector_space K V] (p : V → Prop) :=
+    (p_zero : p 0)
+    (p_add : ∀ u v, p u → p v → p (u + v))
+    (p_neg : ∀ v, p v → p (-v))
+    (p_smul : ∀ c v, p v → p (c • v))
+
+namespace subspace
+
+variables (K : Type) {V : Type} [field K] [vector_space K V]
+{p : V → Prop} [subspace K V p]
```

Comment:
Style: indent the second line, either two spaces or 10 (to line up with the first binder), or use a second `variables` command.